### PR TITLE
Allow id, name classes to headings, plus additional button classes in ckeditor (Closes #762)

### LINKED ISSUE #762: Allow id in headings on CKEditor
## Motivation
I want to be able to add anchor links in the Standard text format

## Solution
Add ` id="" name=""` to the following tags

- h1
- h2
- h3
- h4
- h5
- h6
- div
- span

### PR DESCRIPTION
**Pull request by smayden**
This PR makes "name" and "id" allowable for headings, divs, and spans, and also expands the allowable button classes

## Description
This PR changes the allowed html for the standard text format modules/custom/az_media/config/install/filter.format.az_standard.yml

Specifically adding `name=""` and `id=""` to heading, div, and span tags. This will allow content editors to create anchor links pointing to ids on these tags.

I also went ahead and added the additional button stylings that @barrettbaffert mentions on [this ticket](https://github.com/az-digital/arizona-gatsby/issues/35) for the Admissions project. This allows more advanced editors to add button classes in the source for:
`btn-block btn-lg btn-sm btn-arrow`

image

## Related Issue
#762 
Also see: https://github.com/az-digital/arizona-gatsby/issues/35

## How Has This Been Tested?
Tested locally and in Probo

To test:

1. Create a page
2. Add a paragraph using the standard text editor
3. Add headings and text (can also test spans and divs)
3. View Source
4. Add name and id to heading (or span/div) tags
5. Save page and then edit again to verify that name and id not stripped from source code
6. Create an anchor link pointing to one or more of your ids
7. Verify anchor link works
8. Add buttons and apply button classes in the source code (btn-lg btn-sm btn-block btn-arrow)
9. Verify buttons appear as expected after saving

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_media/config/install/filter.format.az_standard.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ filters:
...
...s> <abbr> <address> <cite> <blockquote class="blockquote blockquote-reverse text-right text-center"> <br> <div class="col col-* row" name="" id=""> <ul type class="nav nav-* ul-*"> <ol start type> <li> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table> <caption> <tbody> <thead> <tfoot> <th colspan="" rowspan=""> <td colspan="" rowspan=""> <tr colspan="" rowspan=""> <hr> <a href class="btn btn-red btn-outline-red btn-blue btn-outline-blue btn-outline-white btn-link btn-block btn-lg btn-sm btn-arrow"> target="" name="" id=""> <span class="text-* badge badge-* small lead h1 h2 h3 h4 h5 h6 align-* m* p*" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <div class=* data-* > <p class="text-align-left text-align-center text-align-right"> <footer class="blockquote-footer">'
```
Comment (danahertzberg) [inline modules/custom/az_media/config/install/filter.format.az_standard.yml:24]: I recommend we get rid of the specific designation of classes to allow all classes on the `<a>` tag.

```suggestion
      allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class="blockquote blockquote-reverse text-right text-center"> <br> <div class="col col-* row" name="" id=""> <ul type class="nav nav-* ul-*"> <ol start type> <li> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table> <caption> <tbody> <thead> <tfoot> <th colspan="" rowspan=""> <td colspan="" rowspan=""> <tr colspan="" rowspan=""> <hr> <a href class="" target="" name="" id=""> <span class="text-* badge badge-* small lead h1 h2 h3 h4 h5 h6 align-* m* p*" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <div class=* data-* > <p class="text-align-left text-align-center text-align-right"> <footer class="blockquote-footer">'
```

This is probably a separate conversation. But do we want all tags to allow all classes? Like add `class=""` to `<span>` `<p>` `<hr>` and `<div>`?

**Comment by danahertzberg** [comment]
This is probably a separate conversation. But do we want all tags to allow all classes? Like add class="" to `<span>` `<p>` `<hr>` and `<div>`?

File: modules/custom/az_media/config/install/filter.format.az_standard.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ filters:
...
+      allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class="blockquote blockquote-reverse text-right text-center"> <br> <div class="col col-* row" name="" id=""> <ul type class="nav nav-* ul-*"> <ol start type> <li> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table> <caption> <tbody> <thead> <tfoot> <th colspan="" rowspan=""> <td colspan="" rowspan=""> <tr colspan="" rowspan=""> <hr> <a href class="" target="" name="" id=""> <span class="text-* badge badge-* small lead h1 h2 h3 h4 h5 h6 align-* m* p*" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <div class=* data-* > <p class="text-align-left text-align-center text-align-right"> <footer class="blockquote-footer">'
```
Comment (danahertzberg) [inline modules/custom/az_media/config/install/filter.format.az_standard.yml:24]:
```suggestion
      allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class=""> <br> <div class="" name="" id="" class=* data-*> <ul type class=""> <ol start type class=""> <li class=""> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table class=""> <caption> <tbody class=""> <thead> <tfoot> <th colspan="" rowspan=""> <td colspan="" rowspan=""> <tr colspan="" rowspan=""> <hr class=""> <a href class="" target="" name="" id=""> <span class="" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <p class=""> <footer class="">'
```

Comment (danahertzberg) [reply modules/custom/az_media/config/install/filter.format.az_standard.yml:24]: `` is also allowed here, but the `` tag is _not_ allowed in QS1

Comment (smayden) [reply modules/custom/az_media/config/install/filter.format.az_standard.yml:24]: @danahertzberg For the div, is it ok that class is listed twice? 
`<div class="" name="" id="" class=* data-*>`

**Review by danahertzberg — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Pull request merged**